syntax = "proto3";

package state;

service StateMachine{

    rpc trigger(TriggerRequest) returns (TriggerResponse) {}

    rpc getState(VoidRequest) returns (StateResponse) {}

}

message StateResponse{
    string state = 1;
}

message TriggerRequest{
    string state = 1;
}

message TriggerResponse{
    bool success = 1;
}

message VoidResponse{}

message VoidRequest{}